member.php

```diff
@@ -1598,11 +1599,30 @@
 
 		if($password_length < 8)
 		{
-			$password_length = 8;
+			$password_length = min(8, (int)$mybb->settings['maxpasswordlength']);
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';

```

Comment:
Shouldn't matter here, but as we use it everywhere else too, it should be changed. @Sama34